### PR TITLE
Added python pitfalls docs and some corrections in other parts

### PR DESCRIPTION
**Pull request by andrptrc**

### CODE REVIEW

**Comment by andrptrc** [comment]
@rmarquis If you have other pitfalls I forgot to mention, feel free to add them !

**Comment by andrptrc** [comment]
Closes #48

File: docs/explanations/about-python-pitfalls.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## How and why do we use Python
+
+The [Core engine](../reference/core-engine.md) and all the services are
+developed using Python.
+
+## Install Python
+
+Download the installation package for your operating system on the official
+[Download center](https://www.python.org/downloads/) and install it.
```
Comment (rmarquis) [inline docs/explanations/about-python-pitfalls.md:38]: Do we need to mention Python 3.11 (and not the latest) should be installed?

As general note, handling different Python versions on the same machine is very error prone, and usually achieved in a different manner from one OS to another (pyenv is usually used on Unix, but doesn't really exist on Windows).

The less worse option I found (rye) is an experimental tool - it still gives me headache sometimes.

**Review by rmarquis — APPROVED** [review]
LGTM. See comment on Python version.

File: docs/explanations/about-python-pitfalls.md (hunk 1 above)
Comment (andrptrc) [reply docs/explanations/about-python-pitfalls.md:38]: Yes, we can mention the version we use, but explaining in detail who to install it on each OS is not part of our work I think.

**Pull request merged**